pkg/storage/redis.go

```diff
@@ -81,9 +151,19 @@ func (b *RedisDB) WriteMany(ctx context.Context, namespaces, keys []string, valu
 	return b.db.MSet(ctx, nameSpaceKeys, values, 0).Err()
 }
 
-func (b *RedisDB) Read(ctx context.Context, namespace, key string) ([]byte, error) {
+func (b *RedisDB) Read(ctx *context.Context, namespace, key string) ([]byte, error) {
 	nameSpaceKey := getRedisKey(namespace, key)
-	res, err := b.db.Get(ctx, nameSpaceKey).Bytes()
+
+	if (*ctx).Value(TX) != nil {
+		txContext := (*ctx).Value(TX).(TxContext)
+
+		watchKeys := txContext.watchKeys
+		watchKeys = append(watchKeys, nameSpaceKey)
+
+		*ctx = context.WithValue(*ctx, TX, TxContext{txContext.tx, watchKeys, txContext.watchOnly})
+	}
```

Comment:
From the [godoc](https://pkg.go.dev/context), "Use context Values only for request-scoped data that transits processes and APIs, not for passing optional parameters to functions."

The recommendation from that doc is to follow those guidelines, which I agree with, so we get the benefits of static analysis.

I believe it's possible to do this by modifying the API. For example, you can have:
* A separate struct where you accumulate the things that need to be watched.
* A separate set of Read/Write functions to be used when you're doing transactions.

For the second part, I think it's ok (and even expected), that callers of the API should know what they're doing when executing transactions. 

Happy to take a stab at that together.

Reply:
Yea I'll change it up so that we have different read/write functions for tx. Was trying to force it in to be plug and play but I guess its ok if we have 2 different read/write